### PR TITLE
Continued delete implementation

### PR DESCRIPTION
**Pull request by natashawong**
Finished implementation + tested up till engagement level. Code implemented but not tested at engagement level (i.e. deleting an engagement) because blocked by Matt working on backend. Receiving error 500 when trying to create a new engagement.

### CODE REVIEW

File: pages/api/deleteEngagement.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { deleteEngagement } from "../../util/deleteEntry";
+
+export default async function handler(req, res) {
+    console.log(req?.body)
+    if (req.method !== 'POST') {
+      res.status(405).send({ message: 'Only POST requests allowed' })
+    }
+    try {
+        console.log(req.body);
+        const response = deleteEngagement(req.body);
```
Comment (Jkwok2018) [inline pages/api/deleteEngagement.js:10]: need to add **await** before call so that the page refreshes successfully (i.e. 
`const response = **await** deleteEngagement(req.body);`
). Should do it for all the other delete endpoints

Comment (Jkwok2018) [inline pages/api/deleteEngagement.js:10]: need to add **await** before call so that the page refreshes successfully (i.e. 
`const response = **await** deleteEngagement(req.body);`
). Should do it for all the other delete endpoints

**Review by Jkwok2018 — CHANGES_REQUESTED** [review]
Delete BOM APIs are not working yet. FYI each device in BOM now has an uniquely identified key `_id` instead of `deviceId`. Also when deleting device from test case BOM, need to remember to update the summaryBOM by finding the new maximum quantity for that device (probably can reuse code from editBOMDevice).
